d2/src/main/java/com/linkedin/d2/balancer/servers/ZooKeeperAnnouncer.java

```diff
@@ -724,6 +724,10 @@ public void setEventEmitter(ServiceDiscoveryEventEmitter emitter) {
 
   @Override
   public void emitSDStatusActiveUpdateIntentAndWriteEvents(String cluster, boolean isMarkUp, boolean succeeded, long startAt) {
+    if (!(_server instanceof ZooKeeperServer))
```

Comment:
when _server is IndisZookeeperServer, it needs to emit the events too. That's why I mentioned "should happen only when _server is NOT an instance of IndisServer (which means it only emits the event when it's doing zk-only or dual write)"